### PR TITLE
Removed the direct inclusion of Pry and make it conditional based on the ENABLE_PRY environment variable

### PR DESCRIPTION
**Pull request by AlajeBash**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-ruby-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
I update all the configuration files by removing Pry from the default setup and add a conditional configuration.
### Debugging with Pry

To Enable Pry for debugging, follow these steps:
1. Open a terminal in your project directory.
2. Set the `ENABLE_PRY` environment variable by running the following command:
   ```bash
   export ENABLE_PRY=1
   ```
This command enables Pry for the current session.

### Debugging without Pry

To Disable Pry for debugging, follow these steps:
Set the `ENABLE_PRY` environment variable by running the following command:
   ```bash
    unset ENABLE_PRY
   ```
*Note:* Pry is included only when the ENABLE_PRY environment variable is set. When it's not set, Pry will not be part of your project's default setup

Submitter Checklist:
- [x] Include a link to the related GitHub issue, [ Link to the problem #2262
- [ ] Include a security review link, if applicable

# Testing
The agent includes a suite of unit and functional tests which should be used to
verify your changes don't break existing functionality. These tests will run with 
GitHub Actions when a pull request is made. More details on running the tests locally can be found 
[here for our unit tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/README.md), 
and [here for our functional tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/multiverse/README.md).
For most contributions it is strongly recommended to add additional tests which
exercise your changes. 

# Reviewer Checklist
- [x] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

### CODE REVIEW

**Comment by kaylareopelle** [comment]
Hi @AlajeBash! Thanks for your submission! 

Remember to [sign the CLA](https://github.com/newrelic/newrelic-ruby-agent/pull/2265#issuecomment-1765495062) so we'll be able to accept it! 😄 

It looks like our Rubocop linter found a few issues:
```
Offenses:

test/environments/norails/Gemfile:20:3: C: [Correctable] Layout/BlockAlignment: end at 20, 2 is not aligned with group :development do at 15, 0.
  end
  ^^^
test/multiverse/suites/active_record_pg/active_record_test.rb:5[9](https://github.com/newrelic/newrelic-ruby-agent/actions/runs/6541169160/job/17792085637?pr=2265#step:6:10)7:7: W: Lint/Debugger: Remove debugger entry point binding.pry.

      binding.pry
      ^^^^^^^^^^^
```

Once those are fixed, we'll be able to proceed with the review.

File: test/multiverse/suites/active_record_pg/active_record_test.rb [hunk 1]
```diff
@@ -591,8 +591,11 @@ def adapter
     # ActiveRecord::Base.configurations[NewRelic::Control.instance.env]['adapter']
     #       ActiveRecord::Base.configs_for(env_name: NewRelic::Control.instance.env)['adapter']
     #       ActiveRecord::Base.configs_for(env_name: RAILS_ENV)['adapter']
-    # require 'pry'; binding.pry
     # Order.configurations[RAILS_ENV]['adapter']
+    if ENV['ENABLE_PRY']
+      require 'pry'
+      binding.pry
+    end
```
Comment (kaylareopelle) [inline test/multiverse/suites/active_record_pg/active_record_test.rb:598]: I don't think we want to preserve this `pry` reference. My guess is that a PR was accidentally merged with some commented-out, debugging-related code.

**Comment by AlajeBash** [comment]
@kaylareopelle thank you for your and feedback, i will fix the offenses and sign the CLA.

File: test/multiverse/suites/active_record_pg/active_record_test.rb (hunk 1 above)
Comment (AlajeBash) [reply test/multiverse/suites/active_record_pg/active_record_test.rb:598]: Ok i removed the `pry` reference from the file

**Comment by AlajeBash** [comment]
@kaylareopelle i'm waiting for your review. Thanks

**Pull request merged**